modules/cockroachdb/cockroachdb_test.go

```diff
@@ -18,26 +15,28 @@ import (
 )
 
 func TestCockroach_Insecure(t *testing.T) {
-	suite.Run(t, &AuthNSuite{
-		url: "postgres://root@localhost:xxxxx/defaultdb?sslmode=disable",
-	})
+	suite.Run(t, &AuthNSuite{})
 }
 
 func TestCockroach_NotRoot(t *testing.T) {
 	suite.Run(t, &AuthNSuite{
-		url: "postgres://test@localhost:xxxxx/defaultdb?sslmode=disable",
 		opts: []testcontainers.ContainerCustomizer{
 			cockroachdb.WithUser("test"),
+			// Do not run the default statements as the user used on this test is
+			// lacking the needed MODIFYCLUSTERSETTING privilege to run them.
+			cockroachdb.WithStatements(),
 		},
 	})
 }
 
 func TestCockroach_Password(t *testing.T) {
 	suite.Run(t, &AuthNSuite{
-		url: "postgres://foo:bar@localhost:xxxxx/defaultdb?sslmode=disable",
 		opts: []testcontainers.ContainerCustomizer{
 			cockroachdb.WithUser("foo"),
 			cockroachdb.WithPassword("bar"),
+			// Do not run the default statements as the user used on this test is
```

Comment:
question: after reading this comment, is it possible to have a test using `WithUser` and with the default statements?

Reply:
So at that point, the user must provide statements to configure the `MODIFYCLUSTERSETTING` privilege for the user in order this to work, right? Should the library grant this privilege automatically in the case the `WithUser` option is passed?

Reply:
I suppose that could an option, yes. Alternatively, [it looks like](https://github.com/testcontainers/testcontainers-go/blob/main/modules/cockroachdb/cockroachdb.go#L144) we don't allow using the `TLS` option, should we maybe do the same here?

Reply:
We can probably do it in a follow-up. Wdyt? Would you like to be involved in that?

Reply:
I'm happy to be involved, but if the preferred path is to make this error if default statements are used with a non-root user then it might be better to do it in this PR so we don't ship this only to introduce a breaking change to it right after? Although I suppose the cluster would fail to start in that scenario anyways so maybe it's not that much of a breaking change.